core/src/main/java/org/kiwix/kiwixmobile/core/settings/CorePrefsFragment.kt

```diff
@@ -0,0 +1,377 @@
+      Snackbar.make(requireView(), R.string.all_history_cleared, Snackbar.LENGTH_SHORT).show()
+    })
+  }
+
+  private fun showClearAllNotesDialog() {
+    alertDialogShower?.show(KiwixDialog.ClearAllNotes, ::clearAllNotes)
+  }
+
+  private fun clearAllNotes() {
+    if (instance.isExternalStorageWritable) {
+      if (ContextCompat.checkSelfPermission(
+          requireActivity(),
+          Manifest.permission.WRITE_EXTERNAL_STORAGE
+        )
+        != PackageManager.PERMISSION_GRANTED &&
+        sharedPreferenceUtil?.isPlayStoreBuildWithAndroid11OrAbove() == false
+      ) {
+        Snackbar.make(
+          requireView(),
+          R.string.ext_storage_permission_not_granted,
+          Snackbar.LENGTH_SHORT
+        )
+          .show()
+        return
+      }
+      if (File(AddNoteDialog.NOTES_DIRECTORY).deleteRecursively()) {
+        Snackbar.make(requireView(), R.string.notes_deletion_successful, Snackbar.LENGTH_SHORT)
+          .show()
+        return
+      }
+    }
+    Snackbar.make(requireView(), R.string.notes_deletion_unsuccessful, Snackbar.LENGTH_SHORT).show()
+  }
+
+  @SuppressLint("SetJavaScriptEnabled")
+  fun openCredits() {
+    @SuppressLint("InflateParams") val view =
+      LayoutInflater.from(
+        requireActivity()
+      ).inflate(R.layout.credits_webview, null) as WebView
+    view.loadUrl("file:///android_asset/credits.html")
+    if (nightModeConfig?.isNightModeActive() == true) {
+      view.settings.javaScriptEnabled = true
+      view.setBackgroundColor(0)
+    }
+    alertDialogShower?.show(OpenCredits { view })
+  }
+
+  override fun onPreferenceTreeClick(preference: Preference): Boolean {
+    if (preference.key.equals(PREF_CLEAR_ALL_HISTORY, ignoreCase = true)) {
+      clearAllHistoryDialog()
+    }
+    if (preference.key.equals(PREF_CLEAR_ALL_NOTES, ignoreCase = true)) {
+      showClearAllNotesDialog()
+    }
+    if (preference.key.equals(PREF_CREDITS, ignoreCase = true)) {
+      openCredits()
+    }
+    if (preference.key.equals(SharedPreferenceUtil.PREF_STORAGE, ignoreCase = true)) {
+      openFolderSelect()
+    }
+    return true
+  }
+
+  private fun openFolderSelect() {
+    val dialogFragment = StorageSelectDialog()
+    dialogFragment.onSelectAction =
+      ::onStorageDeviceSelected
+    dialogFragment.show(
+      requireActivity().supportFragmentManager,
+      resources.getString(R.string.pref_storage)
+    )
+  }
+
+  @Suppress("NestedBlockDepth")
+  private fun onStorageDeviceSelected(storageDevice: StorageDevice) {
+    sharedPreferenceUtil?.let { sharedPreferenceUtil ->
+      findPreference<Preference>(SharedPreferenceUtil.PREF_STORAGE)?.summary =
+        storageCalculator?.calculateAvailableSpace(storageDevice.file)
+      if (storageDevice.isInternal) {
+        sharedPreferenceUtil.putPrefStorage(
+          sharedPreferenceUtil.getPublicDirectoryPath(storageDevice.name)
+        )
+        findPreference<Preference>(SharedPreferenceUtil.PREF_STORAGE)!!.title =
```

Comment:
Remove the double bang operator. We don't encourage double bang in this project unless we wanna crash if something goes really wrong. That's how we gonna get the crash reports :)

Reply:
hi @gouri-panda  

done